src/components/provider.ts

```diff
@@ -3,13 +3,15 @@ import Connector from './connector';
 import {provide, Injector, Provider, Inject, Injectable} from 'angular2/core';
 import * as Redux from 'redux';
 import {INgRedux} from './interfaces';
+import {NgRedux} from './ng-redux';
 
 export function provider<T>(store: Redux.Store<T>) {
   const _connector = new Connector(store);
   const factory = (): INgRedux<T> => <INgRedux<T>>Object.assign({},
-    { connect: _connector.connect,
+    {
+      connect: _connector.connect,
       mapDispatchToTarget: _connector.mapDispatchToTarget
     },
     store);
-  return provide('ngRedux', { useFactory: factory });
+  return provide(NgRedux, { useFactory: () => new NgRedux(factory()) });
```

Comment:
@wbuchwalter @bennett000

Wasn't a fan of needing to have the mix of `@Inject`/etc - this lets us just deal with registering one provider.